[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave devi…

juhaki opened the pull request:
…ce for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,360 @@
+		for(def ep : 1..numberOfChild) {
+			try {
+				refresh(ep)
+			} catch(Exception e) {
```
what exception are you expecting here?

juhaki replied:
When the refresh function internally have some error.
And then. Exception log export.

greens replied:
I understand that. But have you seen exceptions here before?

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,360 @@
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = "Dawon Smart Switch${endpoint}"
+			def child = addChildDevice("Child Switch", childDni, device.hubId,
+				[completedSetup: true, label: componentLabel, isComponent: false])
```
Can you share a picture of this device on the Jira ticket? I want to know whether or not the child devices should be in the main device page or not.

juhaki replied:
I have upload the product picture to the below tickets.

WWST-4892 : In development - [Powermanager] In-Wall Switch (1 gang)
WWST-4872 : In development - [Powermanager] In-Wall Switch (2 gang)
WWST-4876 : In development - [Powermanager] In-Wall Switch (3 gang)

greens commented:
Did you decide not to do child online/offline?

juhaki commented:
Yes. I had discuss with ChongHoon Lee & Jason Jeong & OY Kwon (WWST) about child online/offline & below specific.
- This device type is Z-wave in-wall switch. And. This device power disconnected by only fuse box in house.
- Child-switch-health.groovy : protocol zigbee defined.
- Child-switch.groovy : Health check does not defined.

After that. I have upload to this git.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,360 @@
+	}
+
+	preferences {
+		input "periodOffset", "number", title: "Reporting period Offset", description: "Adjust reporting period by this many degrees", range: "*..*", displayDuringSetup: false
```
is `periodOffset` meant to be like the `reportingInterval` preference we have in other DTHs? i.e. Is it supposed to adjust how frequently the device reports? If so, the device's `checkInterval` should be based off of this number, probably 2x + 10 minutes.

greens replied:
I'd also recommend renaming it to `periodOffset` if that's what it is, so that we can re-use translation strings.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,360 @@
+ * */
+def ping(endpoint = null) {
+	log.info "ping called: endpoint '${endpoint}' state '${state}'"
+	log.debug "pint : device.currentValue : " + device.currentValue("DeviceWatch-DeviceStatus")
```
Since you're no longer tracking device health for the children, this should be refactored, since you will never call `ping` with an endpoint specified.

greens replied:
So `ping` and `refresh` should probably query each endpoint as well as the temperature and humidity, so that `ping` just becomes a call to `refresh`

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,360 @@
+}
+
+private changeSwitch(endpoint, value) {
+	log.info "changeSwitch called: vlaue: '${value}', endpoint: '${endpoint}'"
```
nit: typo "value"

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki commented:
Request the master branch merge(Pull Request).

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,356 @@
+	}
+
+	preferences {
+		input "periodOffset", "number", title: "Reporting periodOffset", description: "Adjust reporting periodOffset by this many degrees", range: "*..*", displayDuringSetup: false
```
```suggestion
		input "reportingInterval", "number", title: "Reporting interval", description: "How often the device should report in minuted", range: "*..*", displayDuringSetup: false
```

is it minutes? seconds? what are the units being used here?

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,356 @@
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping(endpoint = null) {
```
The way you've written this DTH, `ping` will never be called with an endpoint passed in. And the goal of this method is to determine online/offline status, not to ensure correct state. Therefore only a check of the parent device is necessary.

juhaki committed:
Update devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

Co-Authored-By: Steven Green <steven.green@smartthings.com>

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki commented:
Update done about reportingInterval & Ping function.
Request the master branch merge(Pull Request).

greens reviewed (changes requested):
The name of the file and its containing folder must match.

File structure should be:
`dawon-zwave-wall-smart-switch.src/dawon-zwave-wall-smart-switch.groovy`

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,351 @@
+	}
+
+	preferences {
+		input "reportingInterval", "number", title: "Reporting interval", description: "How often the device should report in minuted", range: "*..*", displayDuringSetup: false
```
I guess I made a typo here, it should say `minutes`. Also is there a default value here?

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,351 @@
+
+def installed() {
+	log.info "Installed called '${device.displayName}'"
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
```
the `checkInterval` value should depend on the `reportingInterval` preference

It should be 2x + 10 minutes

juhaki replied:
The device default value is reportingInterval 120 sec ( 2 min ) , tempOffset 20 , humidityOffset 10%

Do you want to change to the below about checkInterval ? 
( 32 Min -> 14 Min )
sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
==>
sendEvent(name: "checkInterval", value: 2 * 2 * 60 + 10 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])

greens replied:
`sendEvent(name: "checkInterval", value: 2 * reportingInterval + 10 * 60, displayed: false, data:...`

It should depend on the value the user has selected for the preference.

juhaki replied:
The "installed" function patch by reply comment.

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki commented:
Request the master branch merge(Pull Request).

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki commented:
Request the master branch merge(Pull Request).

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki commented:
Request the master branch merge(Pull Request).

File structure patch:
dawon-zwave-wall-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

"installed" function patch.

greens commented on devicetypes/smartthings/dawon-zwave-wall-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,355 @@
+
+def configure() {
+	log.info "configure called"
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
```
you need to use the `reportingInterval` here too or you're just overwriting it.

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.

juhaki commented:
Request the master branch merge(Pull Request).

configure function patch by reportingInterval.

greens commented on devicetypes/smartthings/dawon-zwave-wall-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,359 @@
+	preferences {
+		input "reportingInterval", "number", title: "Reporting interval", description: "How often the device should report in minutes", range: "*..*", displayDuringSetup: false
+		input "tempOffset", "number", title: "Temperature Offset", description: "Adjust temperature by this many degrees", range: "*..*", displayDuringSetup: false
+		input "humidityOffset", "number", title: "Humidity Offset", description: "Adjust humidity by this percentage", range: "*..*", displayDuringSetup: false
```
Since this is actually updating a value on the device I'd prefer if these ranges were actually constrained by what the device can support. I don't want it to be possible to send garbage to the device.

greens replied:
also adding defaults

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device

DawonDNS in-wall switch Z-wave device for Korea.
- default value patch on the preferences input

greens commented on devicetypes/smartthings/dawon-zwave-wall-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,359 @@
+	preferences {
+		input "reportingInterval", "number", title: "Reporting interval", defaultValue: 10, description: "How often the device should report in minutes", range: "*..*", displayDuringSetup: false // default value set to 10 minutes
+		input "tempOffset", "number", title: "Temperature Offset", defaultValue: 2, description: "Adjust temperature by this many degrees", range: "*..*", displayDuringSetup: false // default value 2 ¡Æ
+		input "humidityOffset", "number", title: "Humidity Offset", defaultValue: 10, description: "Adjust humidity by this percentage", range: "*..*", displayDuringSetup: false // default value 10 %
```
please limit the ranges

juhaki replied:
Ok. I will check & update about configuration items range value to DawonDNS company.

juhaki replied:
reportingInterval range is 1~60 minutes.
So. I have update done to reportingInterval.
And. Dawon DNS don't want to tempOffset & humidityOffset configuration items changing for power consumption saving.
So.  tempOffset & humidityOffset configuration items disabled.

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.

juhaki commented:
Request the master branch merge(Pull Request).

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
